infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -41,98 +66,257 @@
 @Slf4j
 public class LogReplicationConfigManager {
 
+    // Represents default client for LR V1, i.e., use case where tables are tagged with
+    // 'is_federated' flag, yet no client is specified in proto
+    private static final String DEFAULT_CLIENT = "00000000-0000-0000-0000-000000000000";
+
     @Getter
     private final CorfuRuntime runtime;
 
-    @Getter
-    private LogReplicationConfig config;
+    private final CorfuStore corfuStore;
+
+    private final String localClusterId;
+
+    private long lastRegistryTableLogTail = Address.NON_ADDRESS;
+
+    private Table<ClientRegistrationId, ClientRegistrationInfo, Message> clientRegistrationTable;
```

Comment:
One more thing Chris, could you please describe the newly added fields (like clientRegistrationTable, registryTableEntries etc) for their usage and need. By reading of this file(LogReplicationConfigManager) multiple times, I couldn't comprehend why all these new fields are needed. May be you have explained this already before. It would make this newly introduced code more readable that way.

Also, please address Medhavi's comment (given offline during standup) on the class level comments on the state hold. Hoping that it would also improve the readability of this file.